### PR TITLE
Fix button bg fill

### LINKED ISSUE #53: Flickering in animation on "See all posts" button
https://user-images.githubusercontent.com/5765650/137633253-14e19981-5e16-49d5-9ddc-95aa3f63d16d.mov

This issue also occurs on the "Meet the team" button, the leftmost "See more" button and the "Apply now!" button on the bottom of the front page.

It seems almost like the issue only occurs on the first button within a component or something?

**My system** 
Google Chrome Version 94.0.4606.81 (Official Build) (x86_64)
MacOS Big Sur 11.06

### PR DESCRIPTION
**Pull request by JoaquinTrinanes**
This PR fixes #53 

I ended up using a custom component that overwrites the `children` props, adding the custom styles and classes. It was the simplest interface I could think of that made sense with our current buttons.

Other changes that I did include:
- Remove ImageContainer CSS module, turn out we can change it using the tailwind config.
- Rename the `useThemeColor` hook to `getThemeColor`. The implementation is exactly the same, but now that it's not a hook we can use it outside React components (in this case, helper functions)
- Removed redundant `loading="eager"` from some images. Our `CustomImage` takes care of that and if we want to change it in the future it will be easier to overwrite.

### CODE REVIEW

File: tailwind.config.js [hunk 1]
```diff
@@ -83,11 +84,6 @@ module.exports = {
         '3xl': '1600px',
         '4xl': '2000px',
       },
-      boxShadow: {
-        'fill-green': `inset 1000px 0 0 0 ${colors.green.DEFAULT}`,
-        'fill-pink': `inset 1000px 0 0 0 ${colors.pink.DEFAULT}`,
-        'fill-pink-dark': `inset 1000px 0 0 0 ${colors.pink.dark}`,
-      },
```
Comment (benrbowers) [inline tailwind.config.js:90]: Was this removed on purpose? Just seems important lol.

File: tailwind.config.js [hunk 2]
```diff
@@ -83,11 +84,6 @@ module.exports = {
         '3xl': '1600px',
         '4xl': '2000px',
       },
-      boxShadow: {
```
Comment (benrbowers) [inline tailwind.config.js:86]: Was this boxShadow removed on purpose? Just seems unrelated to the plugins lol

File: src/components/decoration/buttons/utils.tsx [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  fill,
+  children,
+}) => {
+  const childrenWithProps = React.Children.map(children, (child) => {
+    if (!React.isValidElement(child)) return;
+    const props = child.props;
+    return React.cloneElement(child, {
+      className: classNames(
+        props.className,
+        'hover:!bg-left hover:transition-all duration-500 ease-in'
```
Comment (benrbowers) [inline src/components/decoration/buttons/utils.tsx:34]: Is the "!" in "hover:!bg-left" supposed to be there?

**Review by benrbowers — COMMENTED** [review]
Left a couple comments, but overall looks pretty good.

File: tailwind.config.js (hunk 2 above)
Comment (JoaquinTrinanes) [reply tailwind.config.js:86]: Yes! It was the utility that added the animated background before the fix!

File: src/components/decoration/buttons/utils.tsx (hunk 3 above)
Comment (JoaquinTrinanes) [reply src/components/decoration/buttons/utils.tsx:34]: Yep, it means `! important` in tailwind. Perhaps we can add the `bg-right` instead of the class and avoid it tho 🤔
